### PR TITLE
Streamline setting up the project locally and on CI

### PR DESCRIPTION
**Pull request by mataha**
#### Description
This PR aims to significantly improve setting up the project for both contributors and on GitHub Actions.

#### Motivation and Context
I feel that building the project is not straightforward enough.

#### How Has This Been Tested?
I've run the code snippets included in `CONTRIBUTING.md` on several platforms.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by deathaxe** [comment]
As readme advices creating a virtualenv for building, which is ok, I wonder if it makes sense to include it into the Makefile.

I primary use a local make.cmd batch file to build things, which does exactly this.

**Comment by mataha** [comment]
I'm actually tempted to remove the Makefile - creating a virtualenv and activating it is different for each platform and shell, not to mention extra steps needed to ensure `make` reuses the same instance (I believe `.ONESHELL` is not POSIX?).

I prefer to type the commands manually as I switch between Windows and WSL.
Before, I had a script that called MINGW `make` under the surface:

```cmd
::: make.bat
@mingw32-make --no-print-directory MKDIR="-mkdir 2>nul" RM="-rd /s /q 2>nul" %*
```

But that would not work with virtualenv setup.

**Comment by deathaxe** [comment]
I never used the Makefile either.

**Comment by mataha** [comment]
I've deleted the Makefile and moved the build instructions to README (along with fixing all errors related to imports in `build`).

**Comment by deathaxe** [comment]
I don't see value in changing build scripts beyond probably defaulting non-present options to `True`, which should be achievable by argument definitions.

**Comment by mataha** [comment]
> I don't see value in changing build scripts beyond probably defaulting non-present options to `True`, which should be achievable by argument definitions.

I'm not aware of such feature in `argparse`, how?

**Comment by deathaxe** [comment]
You are probably right with that because those are just flags which store `true` if present.

File: build/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import os
+
+PACKAGE_ROOT = os.path.dirname(os.path.dirname(__file__))
```
Comment (deathaxe) [inline build/__init__.py:3]: Defining a global here and importing it in __main__ causes `python build` to fail. The main purpose of `__main__` is to be able to run the command without the `-m` option.

Comment (mataha) [reply build/__init__.py]: May I interest you in the following code?

```python
if not __package__:
    import os, sys, runpy

    dir, module = os.path.split(os.path.dirname(os.path.realpath(__file__)))
    sys.path[0] = dir
    sys.exit(runpy._run_module_as_main(module, alter_argv=False) and 0)

# rest of __main__.py
```

**Pull request merged**